cli: move URL override logic to a separate package (cli refactor sequence 3/n)

knz opened the pull request:
PR split from https://github.com/cockroachdb/cockroach/pull/82020 to ease review.

knz committed:
cli/client_url: lift flag override in helper function

This is one step towards lifting client_url to a different package.

Release note: None

knz committed:
cli/client_url: lift flag override into helper function (2/)

This is one step towards lifting client_url to a different package.

Release note: None

knz committed:
cli/client_url: lift flag override into helper function (3/)

This is one step towards lifting client_url to a different package.

Release note: None

knz committed:
cli/client_url: lift flag override into helper function (4/)

This is one step towards lifting client_url to a different package.

Release note: None

knz committed:
cli/client_url: lift flag override into helper function (5/)

This is one step towards lifting client_url to a different package.

Release note: None

knz committed:
cli/client_url: lift the config logic into a standlone fn

This is one step towards lifting client_url to a different package.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82070)
<!-- Reviewable:end -->

knz committed:
clientsecopts: use an interface instead of callbacks for AnalyzeClientURL()

This documents the API semantics makes the code slightly easier to
reason about.

Release note: None

knz commented:
the last commit introduces an interface as requested by @otan here https://github.com/cockroachdb/cockroach/pull/82020#discussion_r884207242
